functionary/prompt.py

```diff
@@ -175,3 +175,31 @@ def get_number_of_tokens_of_prefix_assistant(tokenizer: Any) -> int:
     text2 = "<|END_OF_USER|>"
     tok_ids2 = tokenizer.encode(text2, add_special_tokens=False)
     return len(tok_ids1) - len(tok_ids2)
+
+
+def get_chat_template() -> str:
+    chat_template = """{% for message in messages %}
+    {% if message['role'] == 'user' %}
+        {{ message['role'] + ':\n' + message['content'] + '<|END_OF_USER|>' + '\n' }}<br>
```

Comment:
why <br> is needed here?

Reply:
i meant: ```<br>``` (github formatting)

Reply:
Actually we can save the result of this function but it would be hard if we want to customize/change in the future.
I wrote a function to return because to make it clear, the syntax of jinja is quite unfamiliar

Reply:
Oh I add: <br> then remove it later because of the syntax of jinja template is quite crazy, if you add additional space to make it look good, it also adds space to the final result.  without: "<br>" only use: "\n"
when we replace:
chat_template = chat_template.replace("\n", "")
it will replace: "\n" in the template, so I add: "<br>" to only replace "\n" that is next to "<br>"

Reply:
In the other words, `<br>` is to annotate: which "\n" can be removed at the end, to differentiate with: "\n" inside the template

Reply:
Oh I see! gotcha thanks